x-pack/elastic-agent/pkg/agent/control/server/server.go

```diff
@@ -210,70 +213,232 @@ func (s *Server) ProcMeta(ctx context.Context, _ *proto.Empty) (*proto.ProcMetaR
 		specs := sp.Specs()
 
 		for n, spec := range specs {
-			procMeta := &proto.ProcMeta{
-				Name:     n,
-				RouteKey: rk,
-			}
-
-			client := http.Client{
-				Timeout: time.Second * 5,
-			}
 			endpoint := monitoring.MonitoringEndpoint(spec, runtime.GOOS, rk)
-			if strings.HasPrefix(endpoint, "unix://") {
-				client.Transport = &http.Transport{
-					Proxy:       nil,
-					DialContext: socket.DialContext(strings.TrimPrefix(endpoint, "unix://")),
-				}
-				endpoint = "unix"
-			} else if strings.HasPrefix(endpoint, "npipe://") {
-				client.Transport = &http.Transport{
-					Proxy:       nil,
-					DialContext: socket.DialContext(strings.TrimPrefix(endpoint, "npipe:///")),
-				}
-				endpoint = "npipe"
-			}
+			client := newSocketRequester(time.Second*5, n, rk, endpoint)
```

Comment:
@michalpristas Seems some quite fundamental changes here. Can you take a look?

Reply:
I see further down, most of the code is equal but split up in multiple methods which is a nice cleanup.